code/modules/reagents/chemistry/reagents/alchemical_potions.dm

```diff
@@ -0,0 +1,294 @@
+			to_chat(H, "<span class='notice'>You begin to hover gently above the ground...</span>")
+
+/datum/reagent/magic/levitatium/on_mob_end_metabolize(mob/living/carbon/human/H)
+	..()
+	if (istype(H))
+		var/datum/species/S = H.dna.species
+		if(!S.CanFly(H) && (H.movement_type & FLYING))
+			S.toggle_flight(H)
+			to_chat(H, "<span class='notice'>You settle gently back onto the ground...</span>")
+			H.set_resting(FALSE, TRUE)
+
+//	----	ACCELERATIUM	----
+
+/datum/reagent/magic/acceleratium
+	name = "acceleratium"
+	description = "Magic potion that increases the speed of whoever drinks it."
+	color = "#94A77D"
+	metabolization_rate = 1.25 * REAGENTS_METABOLISM
+	
+/datum/reagent/magic/acceleratium/on_mob_metabolize(mob/living/L)
+	..()
+	L.add_movespeed_modifier(type, update = TRUE, priority = 100, multiplicative_slowdown = -1, blacklisted_movetypes = (FLYING|FLOATING))
+
+/datum/reagent/magic/acceleratium/on_mob_end_metabolize(mob/living/L)
+	..()
+	L.remove_movespeed_modifier(type)
+
+//	----	HASTIUM	----
+
+/datum/reagent/magic/levitatium/hastium
+	name = "hastium"
+	description = "Magic potion that increases the speed of whoever drinks it, and grants levitation."
+	color = "#94A77D"
+	metabolization_rate = 1.25 * REAGENTS_METABOLISM
+	
+/datum/reagent/magic/levitatium/hastium/on_mob_metabolize(mob/living/L)
+	..()
+	L.add_movespeed_modifier(type, update = TRUE, priority = 100, multiplicative_slowdown = -1)
+
+/datum/reagent/magic/levitatium/hastium/on_mob_end_metabolize(mob/living/L)
+	..()
+	L.remove_movespeed_modifier(type)
+
+//	----	DRAUGHT OF MIDAS	----
+
+/datum/reagent/magic/midas
+	name = "draught of midas"
+	description = "Magic potion that transforms everything that touches into gold."
+	color = "#FFFF91"
+	taste_description = "the best beer you've ever had"
+
+/datum/reagent/magic/midas/reaction_obj(obj/O, reac_volume)
+	if(ismob(O.loc))
+		return
+	reac_volume = round(reac_volume, 0.1)
+	if(istype(O, /obj/machinery/door) || (O.type!=/obj/structure/mineral_door/gold && istype(O, /obj/structure/mineral_door)))
+		if (reac_volume<MAGIC_REAGENT_TOUCH)
+			return
+		var/obj/structure/mineral_door/D = new /obj/structure/mineral_door/gold(get_turf(O))
+		qdel(O)
+		D.Open()
+	else if (reac_volume>=MAGIC_REAGENT_TOUCH * 0.5)
+		if (O.type == /obj/item/bikehorn)	//bikehorn
+			new /obj/item/bikehorn/golden(get_turf(O))
+		else if (O.type == /obj/item/instrument/violin)
+			new /obj/item/instrument/violin/golden(get_turf(O))
+		else if (istype(O,/obj/item/stack/ore) && O.type!=/obj/item/stack/ore/gold)
+			var/obj/item/stack/ore/catalyst = O
+			new /obj/item/stack/ore/gold(get_turf(O),catalyst.amount)
+		else if (istype(O,/obj/item/stack/sheet/mineral) && O.type!=/obj/item/stack/sheet/mineral/gold)
+			var/obj/item/stack/sheet/mineral/catalyst = O
+			new /obj/item/stack/sheet/mineral/gold(get_turf(O),catalyst.amount)
+		else if (istype(O,/obj/item/clothing/head))
+			new /obj/item/clothing/head/crown(get_turf(O))
+		else
+			return
```

Comment:
Blehh, try making this a proc or something, that's pretty messy. Make sure to use istype() for everything too, not an equality check for O.type